### ISSUE TITLE
Autocomplete for Variable length param of different types

### ISSUE THREAD

**Issue by nfedyashev** [castwide/solargraph#580]
Steps to reproduce:

```
class Foo
  def foo; end
end

class Bar
  def bar; end
end

# @type [Foo, Bar]
obj = create_foo_bar

```

obj's type is properly highlighted as either Foo or Bar.
but in `obj.` autocompletion it seems like always only the first type in variable-length inline type hint is used.

I suspect this one might be rather difficult to implement if possible at all.

What might be the desired behavior instead? Combine and show methods from all of those listed types instead? Display none to enforce more strict/explicit type checks?

**Comment by castwide** [comment]
This one should be resolved by #512 (original issue: #507). I expect to include it in an upcoming release, possibly before the end of the month.

**Comment by nfedyashev** [comment]
Thank you very much!
